server/auth.go

```diff
@@ -475,6 +477,32 @@ func (s *Server) processClientOrLeafAuthentication(c *client) bool {
 		return true
 	}
 
+	if nkey == nil && c.kind == CLIENT && c.opts.Nkey != "" && s.opts.SystemAccount != "" {
+		var nkeyAcc *Account
+		if acc, ok := s.tmpAccounts.Load(c.opts.Nkey); ok {
```

Comment:
Account creation/lookups/updates would need a bit of cleanup, but for background, this map exists because it was possible that during the update/rebuild of an account, it references another account. So we had to "pre-register" an account in that map. We probably should not use it outside of this case.